packages/redux-dynamic-modules/src/ModuleStore.ts

```diff
@@ -1,36 +1,41 @@
-export function configureStore<SagaContext, State>(initialState: DeepPartial<State>, context: SagaContext, ...initialModules: IModule<any>[]): IModuleStore<State>;
-export function configureStore<SagaContext, State>(initialState: DeepPartial<State>, context: SagaContext, ...initialModules: IModule<any>[]): IModuleStore<State> {
-  // setup the saga middleware
-  const sagaMonitor = window["__SAGA_MONITOR_EXTENSION__"] || undefined;
-  const sagaMiddleware: SagaMiddleware<SagaContext> = createSagaMiddleware<any>(
-    {
-      context,
-      sagaMonitor
-    }
+export function configureStore<SagaContext, S1>(initialState: DeepPartial<S1>, plugins: IPlugin[], reduxModule: IModule<S1>): IModuleStore<S1>;
+export function configureStore<SagaContext, S1, S2>(initialState: DeepPartial<S1 & S2>, plugins: IPlugin[], m1: IModule<S1>, m2: IModule<S2>): IModuleStore<S1 & S2>;
+export function configureStore<SagaContext, S1, S2, S3>(initialState: DeepPartial<S1 & S2 & S3>, plugins: IPlugin[], m1: IModule<S1>, m2: IModule<S2>, m3: IModule<S3>): IModuleStore<S1 & S2 & S3>;
+export function configureStore<SagaContext, S1, S2, S3, S4>(initialState: DeepPartial<S1 & S2 & S3 & S4>, plugins: IPlugin[], m1: IModule<S1>, m2: IModule<S2>, m3: IModule<S3>, m4: IModule<S4>): IModuleStore<S1 & S2 & S3 & S4>;
+export function configureStore<SagaContext, S1, S2, S3, S4, S5>(initialState: DeepPartial<S1 & S2 & S3 & S4 & S5>, plugins: IPlugin[], m1: IModule<S1>, m2: IModule<S2>, m3: IModule<S3>, m4: IModule<S4>, m5: IModule<S5>): IModuleStore<S1 & S2 & S3 & S4 & S5>;
+export function configureStore<SagaContext, S1, S2, S3, S4, S5, S6>(initialState: DeepPartial<S1 & S2 & S3 & S4 & S5 & S6>, plugins: IPlugin[], m1: IModule<S1>, m2: IModule<S2>, m3: IModule<S3>, m4: IModule<S4>, m5: IModule<S5>, m6: IModule<S6>): IModuleStore<S1 & S2 & S3 & S4 & S5 & S6>;
+export function configureStore<SagaContext, S1, S2, S3, S4, S5, S6, S7>(initialState: DeepPartial<S1 & S2 & S3 & S4 & S5 & S6 & S7>, plugins: IPlugin[], m1: IModule<S1>, m2: IModule<S2>, m3: IModule<S3>, m4: IModule<S4>, m5: IModule<S5>, m6: IModule<S6>, m7: IModule<S7>): IModuleStore<S1 & S2 & S3 & S4 & S5 & S6 & S7>;
+export function configureStore<SagaContext, S1, S2, S3, S4, S5, S6, S7, S8>(initialState: DeepPartial<S1 & S2 & S3 & S4 & S5 & S6 & S7 & S8>, plugins: IPlugin[], m1: IModule<S1>, m2: IModule<S2>, m3: IModule<S3>, m4: IModule<S4>, m5: IModule<S5>, m6: IModule<S6>, m7: IModule<S7>, m8: IModule<S8>): IModuleStore<S1 & S2 & S3 & S4 & S5 & S6 & S7 & S8>;
+export function configureStore<SagaContext, State>(initialState: DeepPartial<State>, plugins: IPlugin[], ...initialModules: IModule<any>[]): IModuleStore<State>;
+export function configureStore<SagaContext, State>(initialState: DeepPartial<State>, plugins: IPlugin[], ...initialModules: IModule<any>[]): IModuleStore<State> {
+  if (!plugins) {
+    plugins = [];
+  }
+
+  const pluginMiddleware = plugins.reduce(

```

Comment:
pluginMiddlewares?

Reply:
Should we also dedupe if any?

Reply:
Renamed to extensions, will rename the middlewares